lib/quantum/supervisor.ex

```diff
@@ -18,11 +18,14 @@ defmodule Quantum.Supervisor do
     opts = quantum_init(quantum, opts)
 
     children = [
+      worker(Quantum.TaskRegistry, [opts], restart: :permanent),
+      worker(Quantum.JobBroadcaster, [opts], restart: :permanent),
+      worker(Quantum.ExecutionBroadcaster, [opts], restart: :permanent),
+      supervisor(Quantum.ExecutorSupervisor, [opts], restart: :permanent),
       supervisor(Task.Supervisor, [[name: Keyword.get(opts, :task_supervisor)]]),
-      worker(Quantum.Runner, [opts], restart: :permanent)
     ]
 
-    supervise(children, strategy: :one_for_one)
+    supervise(children, strategy: :rest_for_one)
```

Comment:
Is `rest_for_one` necessary? What happens when an upstream dies in a gen_stage chain?